src/php/lib/Grpc/Interceptor.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+namespace Grpc;
+
+/**
+ * Represents an interceptor that intercept operations for 4 methods before call starts.
+ */
+class Interceptor
+{
+    public function interceptUnaryUnary(
+      $method,
+      $argument,
+      $deserialize,
+                             array $metadata = [],
+      array $options = [],
+      $continuation
+  ) {
```

Comment:
formatting: is this the appropriate indentation? I have seen it elsewhere in other files as well.

Additionally, I think this file needs a bit more formatting care (I will skip reviewing formatting for now as it needs reformatting).